python/mxnet/contrib/onnx/mx2onnx/_op_translations.py

```diff
@@ -3275,3 +3275,77 @@ def convert_gather_nd(node, **kwargs):
     ]
 
     return nodes
+
+
+@mx_op.register("batch_dot")
+def convert_batch_dot(node, **kwargs):
+    """Map MXNet's batch_dot operator attributes to onnx's operator.
+    """
+    from onnx.helper import make_node
+    name, input_nodes, attrs = get_inputs(node, kwargs)
+
+    lhs = input_nodes[0]
+    rhs = input_nodes[1]
+    input_type = kwargs['in_type']
+    dtype = onnx.mapping.TENSOR_TYPE_TO_NP_TYPE[input_type]
+    transpose_a = str(attrs.get('transpose_a', 'False'))
+    transpose_b = str(attrs.get('transpose_b', 'False'))
+    perm = [0, 2, 1]
+
+    nodes = [
+        create_tensor([-2], name+'_-2', kwargs['initializer']),
+        create_tensor([-1], name+'_-1', kwargs['initializer']),
+        create_tensor([0], name+'_0', kwargs['initializer']),
+        create_tensor([2], name+'_2', kwargs['initializer']),
+        create_tensor([100], name+'_100', kwargs['initializer']),
+        create_tensor([0], name+'_0f', kwargs['initializer'], dtype=dtype),
+    ]
+
+    if transpose_a == 'False':
+        nodes += [
+            make_node('Add', [lhs, name+'_0f'], [name+'_lhs']),
```

Comment:
can we get rid of this add by writing four `make_node('MatMul', [name+'_lhs', name+'_rhs'], [name])` in each case? this drags performance down